feature/agenda/presentation/src/main/java/com/sergius/agenda/presentation/task/TaskDetailsState.kt

```diff
@@ -0,0 +1,32 @@
+package com.sergius.agenda.presentation.task
+
+import androidx.compose.material3.DatePickerState
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.TimePickerState
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+@OptIn(ExperimentalMaterial3Api::class)
+data class TaskDetailsState(
+    val taskId: String? = null,
+    val taskTitle: String = "",
+    val taskDescription: String = "",
+    val showTimerDialog: Boolean = false,
+    val timePickerState: TimePickerState = TimePickerState(initialHour = 0, initialMinute = 0, is24Hour = true),
+    val showDateDialog: Boolean = false,
+    val selectedDate: Long? = null,
+    val datePickerState: DatePickerState = DatePickerState(locale = Locale.getDefault())
+)
+
+@OptIn(ExperimentalMaterial3Api::class)
+fun TimePickerState.toFormattedTime(): String {
+    val hour = if (hour < 10) "0$hour" else hour.toString()
+    val minute = if (minute < 10) "0$minute" else minute.toString()
+    return "$hour:$minute"
+}
+
+fun Long.convertMillisToDate(): String {
+    val formatter = SimpleDateFormat("MMM dd, yyyy", Locale.getDefault())
+    return formatter.format(Date(this))
```

Comment:
The more modern way to format dates is by using the Java LocalDateTime API, since `SimpleDateFormat` is not thread-safe (not an issue here, but as soon as you'd share this `formatter` instance across threads you can run into issues). 
```
val dateTime = ZonedDateTime.ofInstant(
    Instant.ofEpochMilli(millis),
    ZoneId.of("UTC")
)
val formattedDateTime = DateTimeFormatter
    .ofPattern("dd MMM yyyy")
    .format(dateTime)
```